infrastructure/src/main/java/org/corfudb/infrastructure/orchestrator/workflows/AddNode.java

```diff
@@ -0,0 +1,79 @@
+package org.corfudb.infrastructure.orchestrator.workflows;
+
+import lombok.Getter;
+import org.corfudb.infrastructure.orchestrator.Action;
+import org.corfudb.infrastructure.orchestrator.Workflow;
+import org.corfudb.runtime.CorfuRuntime;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ *
+ * A workflow that adds a new node to the cluster.
+ *
+ * Created by Maithem on 10/25/17.
+ */
+
+public class AddNode extends Workflow {
+
+    @Getter
+    String endpoint;
+
+    public AddNode(String endpoint) {
+        this.endpoint = endpoint;
+    }
+
+    @Override
+    public String getName() {
+        return this.getClass().getName();
```

Comment:
Avoid the use of reflection to do this, there seems to be a significant performance penalty. Refactoring tools should be able to handle name changes...

Reply:
I dont think this is a real performance problem due to the frequency of workflow execution. Seems convenient, as opposed to having a string literal. Probably should use the Enum? since enums will map 1-to-1 with these classes.

Reply:
Why call a method when you can put a string literal once?

Reply:
since enums will map 1-to-1 with these classes, i will just use the enums instead.